pkg/loki/modules.go

```diff
@@ -750,7 +750,8 @@ func (t *Loki) initCompactor() (services.Service, error) {
 	t.Server.HTTP.Path("/compactor/ring").Methods("GET", "POST").Handler(t.compactor)
 
 	// TODO: update this when the other deletion modes are available
-	if t.Cfg.CompactorConfig.RetentionEnabled && t.compactor.DeleteMode() == deletion.WholeStreamDeletion {
+	deleteMode := t.compactor.DeleteMode()
+	if t.Cfg.CompactorConfig.RetentionEnabled && (deleteMode == deletion.WholeStreamDeletion || deleteMode == deletion.FilterAndDelete) {
```

Comment:
Same here. The query frontend always needs this API unless deletes are disabled altogether

Reply:
Updated to use the switch statement as in https://github.com/grafana/loki/pull/5922/files